core/deployment/src/main/java/io/quarkus/deployment/dev/testing/TestConfig.java

```diff
@@ -45,6 +47,15 @@ public interface TestConfig {
     @WithDefault("false")
     boolean displayTestOutput();
 
+    /**
+     * The FQN of the JUnit ClassOrdered to use. If the class cannot be found, it defaults backs to JUnit class order.
+     *
+     * @see <a href=https://junit.org/junit5/docs/current/user-guide/#writing-tests-test-execution-order-classes>JUnit Class
+     *      Order<a/>
+     */
+    @WithDefault("io.quarkus.test.junit.util.QuarkusTestProfileAwareClassOrderer")
+    Optional<String> classOrderer();
```

Comment:
Would our JUnit extension continue to work if this is changed?

Reply:
Yes. The `QuarkusTestProfileAwareClassOrderer` is used to optimize the test run order and minimize Quarkus restarts when executing tests: https://github.com/quarkusio/quarkus/pull/20156

It is also documented here:
https://quarkus.io/version/3.15/guides/getting-started-testing#testing_different_profiles

Users can replace ours and provide their own, but setting that up is a bit cumbersome (JUnit only supports a single config file, which we provide and you have to exclude dependencies).

What I did is to provide a global empty one that can use our configuration to delegate to it:
https://github.com/quarkusio/quarkus/pull/42715/files#diff-8f3e9670c7afd7364f77c048342c4973cab00e10963299434315f736f348ddc8R13-R54

By default, you get `QuarkusTestProfileAwareClassOrderer`. To get rid of it or override it, we only need to set `quarks.test.class-orderer`. This is actually a breaking change.

Reply:
Right. What I am worried about is that if users use their own, nothing will work - which is unexpected and we should definitely tell users what to do.

Reply:
Yes, we need to add a note to use `quarks.test.class-orderer` instead.

Reply:
👍🏽